src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp

```diff
@@ -137,960 +112,865 @@ enum XT002Data
- *
- *        XE-321 BOOMBOT
- *
- *///----------------------------------------------------
 class BoomEvent : public BasicEvent
 {
-    public:
-        BoomEvent(Creature* me) : _me(me)
-        {
-        }
+public:
+    BoomEvent(Creature* me) : _me(me) { }
 
-        bool Execute(uint64 /*time*/, uint32 /*diff*/) override
-        {
-            // This hack is here because we suspect our implementation of spell effect execution on targets
-            // is done in the wrong order. We suspect that EFFECT_0 needs to be applied on all targets,
-            // then EFFECT_1, etc - instead of applying each effect on target1, then target2, etc.
-            // The above situation causes the visual for this spell to be bugged, so we remove the instakill
-            // effect and implement a script hack for that.
-
-            _me->CastSpell(_me, SPELL_BOOM, false);
-            return true;
-        }
+    bool Execute(uint64 /*time*/, uint32 /*diff*/) override
+    {
+        // This hack is here because we suspect our implementation of spell effect execution on targets
+        // is done in the wrong order. We suspect that EFFECT_0 needs to be applied on all targets,
+        // then EFFECT_1, etc - instead of applying each effect on target1, then target2, etc.
+        // The above situation causes the visual for this spell to be bugged, so we remove the instakill
+        // effect and implement a script hack for that.
+        _me->KillSelf();
+        return true;
+    }
 
-    private:
-        Creature* _me;
+private:
+    Creature* _me;
 };
 
-class npc_boombot : public CreatureScript
+struct npc_boombot : public ScriptedAI
 {
-    public:
-        npc_boombot() : CreatureScript("npc_boombot") { }
-
-        struct npc_boombotAI : public ScriptedAI
-        {
-            npc_boombotAI(Creature* creature) : ScriptedAI(creature)
-            {
-                Initialize();
-                _instance = creature->GetInstanceScript();
-            }
+    npc_boombot(Creature* creature) : ScriptedAI(creature), _instance(creature->GetInstanceScript()) { }
 
-            void Initialize()
-            {
-                _boomed = false;
-            }
+    void Reset() override
+    {
+        DoCastSelf(SPELL_321_BOOMBOT_AURA);
+        me->SetReactState(REACT_PASSIVE);
+        _scheduler.CancelAll();
 
-            void Reset() override
-            {
-                Initialize();
+        if (_instance->GetBossState(BOSS_XT002) != IN_PROGRESS)
+        {
+            me->DespawnOrUnsummon();
+            return;
+        }
 
-                DoCast(SPELL_AURA_BOOMBOT); // For achievement
+        // HACK/workaround:
+        // these values aren't confirmed - lack of data - and the values in DB are incorrect
+        // these values are needed for correct damage of Boom spell
+        me->SetFloatValue(UNIT_FIELD_MINDAMAGE, 15000.0f);
+        me->SetFloatValue(UNIT_FIELD_MAXDAMAGE, 18000.0f);
```

Comment:
any better idea here @ariel- ?

Reply:
Just set the damage modifier and variance in db to come close to that value.

Reply:
@Ovahlord: These data are obtained through sniff (although I'm not sure)

Reply:
I thought the comment text was talking about the dmg values, but I could be wrong:
```cpp
// these values aren't confirmed - lack of data - and the values in DB are incorrect
```

Reply:
Yes tkrokli, damage modifers dont get in sniffs.

Reply:
@Gooyeth , no, damage values cannot be sniffed as you get only the final damage result sent. Armor etc. needs to get guessed and then you need to compare the damage with videos or sniffs to get close to your result.

Reply:
If you can provide me with some data samples (ie armor and damage) I can guesstimate proper DmgMult/DmgVariance values